assets/test/channel_test.js

```diff
@@ -105,6 +105,17 @@ describe("join", () => {
     assert.throws(() => channel.join(), /^Error: tried to join multiple times/)
   })
 
+  it("prevents duplicate join when race condition", () => {
+    channel.join()
```

Comment:
shall we check for `assert.equal(socket.channels.length, 0)`?

Reply:
sure! I'll add a separate test case for this assertion.

Reply:
I just realized this spec is actually not impacting this implementation at all as a `channel` is only pushed to the socket `channels` on initialization... `channel.join` has zero impact on that : )

Also, I realized we're not handling the case for duplicate join properly – working on it!